scene/resources/style_box_flat.cpp

```diff
@@ -497,11 +498,16 @@ void StyleBoxFlat::draw(RID p_canvas_item, const Rect2 &p_rect) const {
 	real_t aa_size_scaled = 1.0f;
 	if (aa_on) {
 		real_t scale_factor = 1.0f;
-		const SceneTree *tree = Object::cast_to<SceneTree>(OS::get_singleton()->get_main_loop());
-		if (tree) {
-			const Window *window = tree->get_root();
-			const Vector2 stretch_scale = window->get_stretch_transform().get_scale();
-			scale_factor = MIN(stretch_scale.x, stretch_scale.y);
+		Ref<TextServer> ts = TextServerManager::get_singleton()->get_primary_interface();
```

Comment:
Since it's in `draw` method, you can use `CanvasItem::get_current_item_drawn()` instead of interacting with `TextServer`.

```c++
	CanvasItem *ci = CanvasItem::get_current_item_drawn();
	if (ci) {
		Viewport *vp = ci->get_viewport();
		if (vp) {
			scale_factor = vp->get_oversampling();
		}
	}
```

Reply:
Perfect, much cleaner! Thank you!